spyder/plugins/completion/languageserver/transport/main.py

```diff
@@ -132,6 +137,23 @@ def restore(self):
     client = LanguageServerClient(zmq_in_port=args.zmq_in_port,
                                   zmq_out_port=args.zmq_out_port)
     client.start()
+
+    def watch_parent_process(pid):
+        # exist when the given pid is not alive
```

Comment:
Let's make this a docstring with the following text:

    Exit when the given pid is not alive.

    Code taken from the Python Language Server project.

And please don't forget to better reference the code taken from other projects for next time,